timesketch/lib/sigma.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2020 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Timesketch Sigma lib functions."""
+
+import os
+import codecs
+import logging
+import yaml
+
+from flask import current_app
+
+import sigma.configuration as sigma_configuration
+
+from sigma.backends import elasticsearch as sigma_es
+from sigma.parser import collection as sigma_collection
+from sigma.parser import exceptions as sigma_exceptions
+
+logger = logging.getLogger('timesketch.lib.sigma')
+
+
+def get_sigma_config_file():
+    """Get a sigma.configuration.SigmaConfiguration object.
+
+        Returns:
+            A sigma.configuration.SigmaConfiguration object
+    """
+    config_file = current_app.config.get('SIGMA_CONFIG')
+
+    if not config_file:
+        raise ValueError(
+            'SIGMA_CONFIG not found in config file')
+
+    if not os.path.isfile(config_file):
+        raise ValueError(
+            'Unable to open file: [{0:s}], it does not exist.'.format(
+                config_file))
+
+    if not os.access(config_file, os.R_OK):
+        raise ValueError(
+            'Unable to open file: [{0:s}], cannot open it for '
+            'read, please check permissions.'.format(config_file))
+
+    with open(config_file, 'r') as config_file:
+        sigma_config_file = config_file.read()
+
+    sigma_config = sigma_configuration.SigmaConfiguration(sigma_config_file)
+
+    return sigma_config
+
+def get_sigma_rules_path():
+    """Get Sigma rules paths.
+
+        Returns:
```

Comment:
returns section here should be indented differently

```
"""Docstring....

Returns:
    Stuff
""""

```

this applies elsewhere too

Reply:
also missing `Raises` section in the docstring